vyper/opcodes.py

```
@@ -169,5 +197,67 @@
     'GOTO': (None, 1, 0, 8)
 }
 
+COMB_OPCODES: OpcodeMap = {**OPCODES, **PSEUDO_OPCODES}
+
+
+def evm_wrapper(fn, *args, **kwargs):
+
+    def _wrapper(*args, **kwargs):
+        global active_evm_version
+        evm_version = kwargs.pop('evm_version', None) or DEFAULT_EVM_VERSION
+        active_evm_version = EVM_VERSIONS[evm_version]
+        try:
+            return fn(*args, **kwargs)
+        finally:
+            active_evm_version = EVM_VERSIONS[DEFAULT_EVM_VERSION]
+
+    return _wrapper
+
+
+def _gas(value: OpcodeValue, idx: int) -> Optional[OpcodeRulesetValue]:
+    gas: OpcodeGasCost = value[3]
+    if isinstance(gas, int):
+        return value[:3] + (gas,)
+    if len(gas) <= idx:
+        return value[:3] + (gas[-1],)
+    if gas[idx] is None:
+        return None
+    return value[:3] + (gas[idx],)
+
+
+def _mk_version_opcodes(opcodes: OpcodeMap, idx: int) -> OpcodeRulesetMap:
+    return dict(
+        (k, _gas(v, idx)) for k, v in opcodes.items()  # type: ignore
+        if _gas(v, idx) is not None
+    )
+
+
+_evm_opcodes: Dict[int, OpcodeRulesetMap] = dict(
+    (v, _mk_version_opcodes(OPCODES, v))
+    for v in EVM_VERSIONS.values()
+)
+_evm_combined: Dict[int, OpcodeRulesetMap] = dict(
+    (v, _mk_version_opcodes(COMB_OPCODES, v))
+    for v in EVM_VERSIONS.values()
+)
+
+
+def get_opcodes() -> OpcodeRulesetMap:
+    return _evm_opcodes[active_evm_version]
+
+
+def get_comb_opcodes() -> OpcodeRulesetMap:
+    return _evm_combined[active_evm_version]
+
 
-COMB_OPCODES: Dict[str, Tuple[Union[int, None], int, int, int]] = {**OPCODES, **PSEUDO_OPCODES}
+def version_check(begin: Optional[str] = None, end: Optional[str] = None) -> bool:
+    assert begin is not None or end is not None
```

Should throw a usage exception of some sort

---

```suggestion
    if begin is None and end is None:
        raise CompilerPanic("Either beginning or end fork ruleset must be set.")
```